lib/rubocop/cop/rails/timeout_from_transaction.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for transactions wrapped in timeouts,
+      # motivated by change in Rails behavior that will
+      # cause exited transactions to be rolled back
+      # rather than committed (pre ActiveRecord 7 behavior).
+      #
+      # @example
+      #   # bad
+      #   Timeout.timeout(10) do
+      #     ApplicationRecord.transaction do
+      #       user.save
+      #     end
+      #   end
+      class TimeoutFromTransaction < Base
+        def_node_matcher :timeout_node?, <<~PATTERN
+          (block
+            (send _ :timeout _)
```

Comment:
Can you replace the receiver with `Timeout` constant instead of any variable? And `Timeout.timeout` method have optional arguments. Can you add them to tests and pass them?

> timeout(sec, klass = nil, message = nil) { |sec| ... }

https://docs.ruby-lang.org/en/3.0/Timeout.html